Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -1,162 +1,204 @@
 
-        /// <inheritdoc/>
-        public async ValueTask<IEnumerable<TItem>> CreateAsync(IEnumerable<TItem> values)
+        /// <inheritdoc />
+        public async ValueTask<TItem?> GetAsync(string id, PartitionKey partitionKey)
         {
-            IEnumerable<Task<TItem>> creationTasks =
-                values.Select(v => CreateAsync(v).AsTask()).ToList();
-
-            _ = await Task.WhenAll(creationTasks).ConfigureAwait(false);
+            _ = id ?? throw new ArgumentNullException(nameof(id));
 
-            return creationTasks.Select(x => x.Result);
-        }
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
 
-        /// <inheritdoc/>
-        public async ValueTask<TItem> UpdateAsync(TItem value)
-        {
-            (bool optimizeBandwidth, ItemRequestOptions options) = RequestOptions;
-            Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+            Container container =
+                await this.containerProvider.GetContainerAsync().ConfigureAwait(false);
 
             ItemResponse<TItem> response =
-                await container.UpsertItemAsync<TItem>(value, value.PartitionKey, options).ConfigureAwait(false);
+                await container.ReadItemAsync<TItem>(id, partitionKey).ConfigureAwait(false);
+
+            TItem item = response.Resource;
 
-            TryLogDebugDetails(_logger, () => $"Updated: {JsonConvert.SerializeObject(value)}");
+            TryLogDebugDetails(
+                this.logger,
+                () => string.Format(
+                    Properties.Resources.ReadInsertNameHere,
+                    JsonConvert.SerializeObject(item)));
 
-            return optimizeBandwidth ? value : response.Resource;
+            return string.IsNullOrEmpty(item.Type) || item.Type == typeof(TItem).Name ? item : null;
         }
 
-        /// <inheritdoc/>
-        public ValueTask DeleteAsync(TItem value) => DeleteAsync(value.Id, value.PartitionKey);
+        /// <inheritdoc />
+        public async ValueTask<IEnumerable<TItem>> GetAsync(
+            Expression<Func<TItem, bool>>? predicate = null,
+            CancellationToken cancellationToken = default) =>
+            await this.EnumerateAsync(predicate, cancellationToken).ToListAsync();
 
-        /// <inheritdoc/>
-        public ValueTask DeleteAsync(string id, string partitionKeyValue = null) =>
-            DeleteAsync(id, new PartitionKey(partitionKeyValue ?? id));
-
-        /// <inheritdoc/>
-        public async ValueTask DeleteAsync(string id, PartitionKey partitionKey)
+        /// <inheritdoc />
+        public async ValueTask<TItem> UpdateAsync(TItem value)
         {
-            ItemRequestOptions options = RequestOptions.Options;
-            Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+            _ = value ?? throw new ArgumentNullException(nameof(value));
```

Comment:
I'm a bit conflicted about these, but I suppose they do make sense. We should probably add unit tests for these cases.

Reply:
Yeah, I put in some of these but I know I didn't catch all of them.